### PR TITLE
Fetch schema using schema api and then post to kinesis

### PR DESCRIPTION
**Pull request by jobinthomasnypl**
Posting to kinesis stream has been moved to lib/streams.js file
Instead of inferring bib/item schema from the MARC json file, now the service will get it from the schema api and use that to encode data and post to kinesis.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,33 +1,240 @@
...
+      .then((all_schemas) => {
+        console.log("Sending all schemas")
+        CACHE[SCHEMA_READING_STREAM] = all_schemas[0]
+        CACHE[SCHEMA_POSTING_STREAM] = all_schemas[1]
+        return Promise.resolve(CACHE)
+      })
+      .catch((e) => reject (e))
+}
+
+var setGlobalAccessTokenAndProcessResources = (function(records, schema_data){
```
Comment (EdwinGuzman) [inline index.js]: This function doesn't seem to be used in this codebase.

Comment (jobinthomasnypl) [inline index.js]: @EdwinGuzman ,

Yup, thanks for pointing that out. Will take that out.

**Comment by jobinthomasnypl** [comment]
Made changes based on standard js feedback. The only thing I skipped is using snake case instead of camel case (personal preference).

**Comment by jobinthomasnypl** [comment]
@EdwinGuzman,

What is the advantage of one over the other? My reason for keeping it in config was just one place to make all changes. However, I am open for change.

**Comment by nonword** [comment]
Agree with @EdwinGuzman suggestion to configure operational mode (bib vs item) via an env variable. The advantage is if you have to fix a bug that effects both lambdas, you only have to patch the bug itself and deploy to both lambdas. If you were checking `config.get('isABib')`, you'd have to patch the bug PLUS remember to correctly modify your config in the right way for each build. It's extra work and eventually one of us is going to accidentally publish isABib=false to the sierra-retriever that processes Bibs. In general if we're deploying the codebase to two different lambdas, the code should be truly identical including config - so there are no gotchas. Differences should be outside the code in env vars.

On that, I'd also propose the env var be something non-boolean like `RETRIEVAL_TYPE=(bib|item)`. That generalizes it in case it needs to retrieve other things down the line.

**Comment by jobinthomasnypl** [comment]
@nonword and @EdwinGuzman ,

That is a good point. I have created a task to capture the change. I have made another small change  in the last checkin. Will you be able to check it and let me know if it makes sense

**Comment by jobinthomasnypl** [comment]
Changes made to use isABib value based on environment variable - RETRIEVAL_TYPE=bib|item.
Based on that, process for bibs/items.

File: index.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ var streams = require('./lib/stream')
 const SCHEMA_READING_STREAM = 'schema_reading_stream'
 const SCHEMA_POSTING_STREAM = 'schema_posting_stream'
 
+const isABib = ((process.env.RETRIEVAL_TYPE === 'bib'))
```
Comment (EdwinGuzman) [inline index.js]: You don't need the parenthesis here.

File: index.js [hunk 3]
```diff
@@ -136,6 +137,7 @@ var avro_decoded_data = function (schema_data, record) {
   const type = avro.parse(schema_data)
   var decoded = new Buffer(record, 'base64')
   var verify = type.fromBuffer(decoded)
+  console.log(decoded)
```
Comment (EdwinGuzman) [inline index.js]: You should also added more information to this log so you know where it's coming from when you see all the logs together.

File: index.js (hunk 2 above)
Comment (jobinthomasnypl) [inline index.js]: @EdwinGuzman,

I think it was there in the old commit, the new one doesn't have that. I will squash my commits next time.

File: index.js (hunk 3 above)
Comment (jobinthomasnypl) [inline index.js]: @EdwinGuzman,

I removed it. That was there mainly for troubleshooting purposes. When I do proper logging, I am going to have logging only where ever it is required.

**Pull request merged**